lib/setup/profile.js

```diff
@@ -77,33 +177,53 @@ var appendScript = function(name, options) {
 module.exports.update = function() {
   var handled = false;
   var changed = false;
+  var appendFile;
 
-  var promise = Promise.map(['.bash_profile', '.zshrc'], function(name) {
-    return appendScript(name);
-  })
-  .each(function(result) {
-    if (result) {
-      handled = true;
-      changed = changed || (result === 'change');
+  var promise = shellInitScriptChooser()
+  .then(function(initScript) {
+    appendFile = initScript;
+
+    if (appendFile === bashProfileFileName) {
+      return Promise.map([bashProfileFileName, '.zshrc'], function(name) {
+        return appendScript(name);
+      })
+      .each(function(result) {
+        if (result) {
+          handled = true;
+          changed = changed || (result === 'change');
+        }
+      });
     }
-  });
 
-  promise = promise.then(function() {
+    else if (appendFile === bashrcFileName) {
+      return appendScript(bashrcFileName)
+      .then(function(result) {
+        if (result) {
+          handled = true;
+          changed = changed || (result === 'change');
+        }
+      });
+    }
+  })
+
+  .then(function() {
     if (!handled) {
-      return appendScript('.bash_profile', { force: true });
+      return appendScript(appendFile, { force: true });
     }
   })
   .then(function(result) {
     if (result) {
       handled = true;
       changed = changed || (result === 'change');
     }
-  });
+  })
 
-  return promise.then(function() {
+  .then(function() {
     if (changed) {
       console.log('%s: %s', chalk.bold.magenta('avn'),
         chalk.bold.cyan('restart your terminal to start using avn'));
     }
   });
+
+  return promise;
```

Comment:
This should not change — the returned promise resolution would occur before the final log message is output (which is not desirable & shouldn't be required for this PR).